packages/docusaurus-plugin-content-docs/src/plugin-content-docs.d.ts

```diff
@@ -132,6 +132,7 @@ declare module '@theme/Seo' {
     readonly description?: string;
     readonly keywords?: readonly string[] | string;
     readonly image?: string;
+    readonly metaTags?: Record<string, string>;
```

Comment:
Honestly, I'm not a fan of using the SEO component for all use-cases but I'd rather keep it small and not increase this API.

Head/helmet is here to handle other use-cases

Also, metas can be property/name (difference matter afaik): 

![image](https://user-images.githubusercontent.com/749374/130093963-6a5df8ca-3465-427d-bda8-098e97264859.png)

Using `Record<string,string>` is confusing, less explicit and can only handle one of those 2 cases so it's an error-prone abstraction.

I'd rather use `<Head>` in BlogPostItem even if this feels duplicate with `<Seo>`. 

Eventually:
- add a generic `children` prop to pass additional head elements inside `<Seo>`?
- let `Seo` accept any `Head` prop and spread them? (it has a `meta` prop btw)

Reply:
Okay cool - just pushed a different implementation based upon your feedback. Does that look kind of like what you had in mind?

Reply:
Yes, that looks better to me :)

Another thing to consider: this comp is used in the paginated list view. 

This means that on this page: https://deploy-preview-5375--docusaurus-2.netlify.app/blog/
We have 3 times those metas being declared in `<Head>`, and the last article of the list "wins".

I'm not sure it's what you want in this case. Unlike structured data, page metas override each others.

We have a `isBlogPostPage` prop on this comp that you could use.
(I don't like it much, it's error-prone as we can see here, and I'd rather use composition and have a dedicated comp for the list blog post item)

Note: I'm not sure Head supports React fragments well 🤪

Reply:
> We have a isBlogPostPage prop on this comp that you could use.

Done!